File: lib/domain/enums/role.dart
enum Role {
  admin(label: 'Administrateur', description: 'Accès complet au système'),
  agent(label: 'Agent de maintenance', description: 'Gestion des interventions et du planning'),
  secretary(label: 'Secrétaire', description: 'Gestion du planning et des réservations');

  final String label;
  final String description;

  const Role({required this.label, required this.description});
}

File: courtcare-web/src/domain/enums/role.ts
export enum Role {
  ADMIN = 'admin',
  AGENT = 'agent',
  SECRETARY = 'secretary'
}

export type RoleKey = keyof typeof Role;

export const roleLabels: Record<Role, string> = {
  [Role.ADMIN]: 'Administrateur',
  [Role.AGENT]: 'Agent de maintenance',
  [Role.SECRETARY]: 'Secrétaire'
};

export const roleDescriptions: Record<Role, string> = {
  [Role.ADMIN]: 'Accès complet au système',
  [Role.AGENT]: 'Gestion des interventions et du planning',
  [Role.SECRETARY]: 'Gestion du planning et des réservations'
};
